src/SchemaRegistry.ts

```diff
@@ -158,6 +164,62 @@ export default class SchemaRegistry {
     return registeredSchema
   }
 
+  private async updateOptionsWithSchemaReferences(
+    options: SchemaRegistryAPIClientOptions | undefined,
+    schema: ConfluentSchema,
+  ) {
+    const helper = helperTypeFromSchemaType(schema.type)
+    const referredSchemas = await this.getReferredSchemas(schema, helper)
+    return helper.updateOptionsFromSchemaReferences(options as ProtocolOptions, referredSchemas)
+  }
+
+  private async getReferredSchemas(
+    schema: ConfluentSchema,
+    helper: SchemaHelper,
+  ): Promise<ConfluentSchema[]> {
+    const referencesSet = new Set<string>()
+    return this.getReferredSchemasRecursive(schema, helper, referencesSet)
+  }
+
+  private async getReferredSchemasRecursive(
+    schema: ConfluentSchema,
+    helper: SchemaHelper,
+    referencesSet: Set<string>,
+  ): Promise<ConfluentSchema[]> {
+    const references = schema.references || []
+
+    let referredSchemas: ConfluentSchema[] = []
+    for (const reference of references) {
+      const schemas = await this.getReferredSchemasFromReference(reference, helper, referencesSet)
+      referredSchemas = referredSchemas.concat(schemas)
+    }
+    return referredSchemas
+  }
+
+  async getReferredSchemasFromReference(
+    reference: ReferenceType,
+    helper: SchemaHelper,
+    referencesSet: Set<string>,
+  ): Promise<ConfluentSchema[]> {
+    const { name, subject, version } = reference
+    const key = `${name}-${subject}-${version}`
+
+    // avoid duplicates
+    if (referencesSet.has(key)) {
```

Comment:
Does this prevent infinite recursion if you have schemas that reference each other?

Reply:
It will because the recursive call is after this condition

Reply:
As I understand schema-registry does not allow circular calls. A schema has to be registered before it can be referred. The schemas is a directed acyclic graph.

Reply:
That makes sense to me.